### ISSUE TITLE
[VL] columnar2Row Assert failure

### ISSUE THREAD

**Issue by FelixYBW** [apache/incubator-gluten#7253]
### Backend

VL (Velox)

### Bug description

Columnar.scala:68
```
case class ColumnarToRowExec(child: SparkPlan) extends ColumnarToRowTransition with CodegenSupport {
  assert(child.supportsColumnar)
```

```
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:208)
	at org.apache.spark.sql.execution.ColumnarToRowExec.<init>(Columnar.scala:68)
	at org.apache.spark.sql.execution.ColumnarToRowExec.copy(Columnar.scala:67)
	at org.apache.spark.sql.execution.ColumnarToRowExec.withNewChildInternal(Columnar.scala:207)
	at org.apache.spark.sql.execution.ColumnarToRowExec.withNewChildInternal(Columnar.scala:67)
	at org.apache.spark.sql.catalyst.trees.UnaryLike.withNewChildrenInternal(TreeNode.scala:1142)
	at org.apache.spark.sql.catalyst.trees.UnaryLike.withNewChildrenInternal$(TreeNode.scala:1140)
	at org.apache.spark.sql.execution.ColumnarToRowExec.withNewChildrenInternal(Columnar.scala:67)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$withNewChildren$2(TreeNode.scala:359)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:82)
	at org.apache.spark.sql.catalyst.trees.TreeNode.withNewChildren(TreeNode.scala:358)
	at org.apache.gluten.planner.plan.GlutenPlanModel$PlanModelImpl$.$anonfun$withNewChildren$1(GlutenPlanModel.scala:93)
	at org.apache.gluten.planner.plan.GlutenPlanModel$PlanModelImpl$.fakeTc(GlutenPlanModel.scala:79)
	at org.apache.gluten.planner.plan.GlutenPlanModel$PlanModelImpl$.withNewChildren(GlutenPlanModel.scala:93)
	at org.apache.gluten.planner.plan.GlutenPlanModel$PlanModelImpl$.withNewChildren(GlutenPlanModel.scala:73)
	at org.apache.gluten.ras.Ras.withNewChildren(Ras.scala:133)
	at org.apache.gluten.ras.memo.Memo$RasMemo$InCusterMemo.prepareInsert(Memo.scala:148)
	at org.apache.gluten.ras.memo.Memo$RasMemo$InCusterMemo.memorize(Memo.scala:174)
	at org.apache.gluten.ras.rule.RuleApplier$EnforcerRuleApplier.$anonfun$apply0$2(RuleApplier.scala:107)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.apache.gluten.ras.rule.RuleApplier$EnforcerRuleApplier.apply0(RuleApplier.scala:104)
	at org.apache.gluten.ras.rule.RuleApplier$EnforcerRuleApplier.apply(RuleApplier.scala:97)
	at org.apache.gluten.ras.rule.RuleApplier$ShapeAwareRuleApplier.apply(RuleApplier.scala:122)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.applyRule(DpPlanner.scala:187)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$applyEnforcerRules$3(DpPlanner.scala:155)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$applyEnforcerRules$3$adapted(DpPlanner.scala:155)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$applyEnforcerRules$2(DpPlanner.scala:155)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$applyEnforcerRules$2$adapted(DpPlanner.scala:155)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$findPaths$3(DpPlanner.scala:178)
	at org.apache.gluten.ras.dp.DpPlanner$ExploreAdjustment.$anonfun$findPaths$3$adapted(DpPlanner.scala:178)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.generic.TraversableForwarder.foreach(TraversableForwarder.scala:38)
	at scala.collection.generic.TraversableForwarder.foreach$(TraversableForwarder.scala:38)
	at scala.collection.mutable.ListBuffer.foreach(ListBuffer.scala:47)
	at scala.collection.TraversableViewLike$Appended.foreach(TraversableViewLike.scala:203)
	at scala.collection.TraversableViewLike$Appended.foreach$(TraversableViewLike.scala:201)
	at scala.collection.SeqViewLike$$anon$2.foreach(SeqViewLike.scala:197)
```

### Spark version

None

**Comment by FelixYBW** [comment]
@zhztheplayer Looks the error is here:
```
          fakeTc {
            c2r.withNewChildren(children)
          }
```

**Comment by FelixYBW** [comment]
The passed Plan:
![image](https://github.com/user-attachments/assets/3cb897dd-1a91-42d0-a9bf-ede66b00be73)

**Comment by FelixYBW** [comment]
Closed because it's a spark bug and fixed in 3.2.1
